### ISSUE TITLE
Make input value check user friendlier

### ISSUE THREAD

**Issue by bahmutov** [cypress-io/cypress#988]
## feature
Cypress 1.1.2

When checking value of input box like this
```html
<input id="foo">
```
we have to use `have.value` which is hard to remember https://docs.cypress.io/guides/references/assertions.html#Value

```js
cy.get('#foo').type('something')
cy.get('#foo').should('have.value', 'something')
```
It would be nice to also detect `input` or `textarea` and allow these assertions in this case

```js
cy.get('#foo').should('equal', 'something')
cy.get('#foo').should('contain', 'something')
```

**Comment by brian-mann** [comment]
I like the idea of us being "smarter" about the assertions but the problem to me is consistency and documentation.

I would prefer to **error** when using these assertions and telling you the "right" one to do. I think there is already a huge amount of confusion as to how to use assertions, which one to use, what the differences are, and then taking assertions that only ever mean a specific thing and making them more broadly applicable (only in Cypress land) will add to the confusion, for sure.

Perhaps re-evaluating how we document and present the assertions would be better.

**Comment by FredyC** [comment]
Um, is it actually possible to do `should('contain', 'asdf')` against input value? I am looking through docs and trying things like `should('value.contain', 'asdf')`, but that checks for a full value.

**Comment by jennifer-shehane** [comment]
@FredyC No, the test cases written above are suggestions for how they would like the API to work.

The valid way to check for an input value would be:
```js
cy.get('input').should('have.value', 'asdf')
```

**Comment by jennifer-shehane** [comment]
@FredyC Ah, I see. I think you may be able to write:

```
cy.get('input').invoke('val').should('contain', 'partial')
```

**Comment by rossanoua** [comment]
Hi @jennifer-shehane and @FredyC and of course everybody! ))
I can't figure out of how to check for partial content of `textarea`.
Maybe anyone can suggest me some example?
Thank you all!

**Comment by Bkucera** [comment]
@rossanoua the code @jennifer-shehane mentioned above should do the trick:
```js
cy.get('textarea').invoke('val').should('contain', 'partial')

```

**Comment by rossanoua** [comment]
@Bkucera 
yes! it's worked!
i was tried this code before, but without success. (so, some where was a mistake made by me)
thanks a lot!

**Comment by Bkucera** [comment]
Partial value,html,text,and id matchers are in #3259

**Comment by DanWahlin** [comment]
Thanks @jennifer-shehane. Just ran into this today. I was actually trying out the exact scenario that @bahmutov proposed at the top of the thread to do a contains() on a textarea. Couldn't get it to work and stumbled across this - thanks for the invoke() suggestion. It does feel like contains() should work here, but it sounds like there are additional things to take into consideration.

**Comment by shahzad-mycroft** [comment]
Thanks, @Bkucera your solution works for me
